python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -799,18 +753,25 @@ def _process_auto_materialize_tick_generator(
 
             # Determine if the most recent tick requires retrying
             retry_tick: Optional[InstigatorTick] = None
-
+            override_evaluation_id: Optional[int] = None
             if latest_tick:
+                can_resume = (
+                    get_current_timestamp() - latest_tick.timestamp
+                ) <= MAX_TIME_TO_RESUME_TICK_SECONDS
+                previous_cursor_written = (
+                    latest_tick.automation_condition_evaluation_id == stored_cursor.evaluation_id
+                )
+
+                if can_resume and not previous_cursor_written:
+                    # if the tick failed before writing a cursor, we don't want to advance the
+                    # evaluation id yet
+                    override_evaluation_id = latest_tick.automation_condition_evaluation_id
```

Comment:
The impact of this logic having some kind of an error would be the evaluation history being messed up in the UI right? No effect on the underlying evaluation logic of AMP?

Just confirming that leaving this line out causes some failure recovery test somewhere to fail

Reply:
that's right -- basically the worst case scenario is that we would write duplicate evaluation rows for the same underlying evaluation (so it might look like we requested an asset multiple times even though we only did it once). this would not impact what actually got kicked off though